### PR TITLE
fix Matching.removeEdgesForNoDerivativeFunctionInputs

### PR DESCRIPTION
**Pull request by hkiel**

### CODE REVIEW

File: Compiler/BackEnd/Matching.mo [hunk 1]
```diff
@@ -5666,7 +5666,7 @@ algorithm
       arrayUpdate(m,idx,row);
       //update mt
       for varIdx in varIdxs loop
-        row := arrayGet(m,varIdx);
+        row := arrayGet(mt,varIdx);
```
Comment (hkiel) [inline Compiler/BackEnd/Matching.mo]: @sjoelund Is this fix correct? From my understanding mt should be updated here, but maybe it is the other way round!?

Comment (hkiel) [reply Compiler/BackEnd/Matching.mo]: @vwaurich Can you have a look please?

Comment (sjoelund) [reply Compiler/BackEnd/Matching.mo]: I'm not sure. I guess it's reasonable to get the value and update it instead of getting some different value...

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2017-10-26_07-49-54](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4816/console).

**Pull request closed without merging**